### PR TITLE
small improvement for better handling of many log files (gh-1265)

### PR DESCRIPTION
**Pull request by sebres**
the FileFilter contains now a (ordered) dict of the log files (instead of list), as discussed in gh-1265

### CODE REVIEW

**Comment by yarikoptic** [comment]
Thanks... Let me just do some benchmarking when I get to the laptop... Still not clear to me if there would be any benefit from ordered dict

**Comment by sebres** [comment]
> benchmarking

Don't forget that a writing resp. modifying of this list (add/del) will occur comparatively rarer as reading (get/iter).

File: fail2ban/server/filter.py [hunk 1]
```diff
@@ -38,6 +38,11 @@
 from .action import CommandAction
 from ..helpers import getLogger
 
+try:
+	from collections import OrderedDict
+except ImportError:
+	OrderedDict = dict
+
```
Comment (yarikoptic) [inline fail2ban/server/filter.py]: http://nbviewer.ipython.org/url/www.onerussian.com/tmp/dics-vs-ordereddict.ipynb
so why should we use slower OrderedDict?

File: ChangeLog [hunk 2]
```diff
@@ -59,6 +59,8 @@ ver. 0.9.4 (2015/XX/XXX) - wanna-be-released
      rest api and web interface (gh-1223)
    * Add *_backend options for services to allow distros to set the default
      backend per service, set default to systemd for Fedora as appropriate
+   * small improvement for better handling of many log files (gh-1265)
+     Thanks @kshetragia
```
Comment (yarikoptic) [inline ChangeLog]: gr... I guess github didn't post my comment from the phone :-(
would be better IMHO to provide a bit more specific than "small improvement" description, e.g.

"Use associative array (dict) for monitored log files to speed up lookup operations. Provides performance improvements while monitoring large number of files ..."

**Comment by kshetragia** [comment]
The issue #1265 was already closed so i'll write here.
Thank you for the fast answer and provided solution.

 I agree with you these issues in filter.py are not a root cause of problem and i mentioned ones as POSSIBLE root cause. But problem is real. I've tried to split f2b jail to four ones and pyinotify was used, but f2b performance still is not enough for big amount of log files.

I try to enable these four jails and f2b client process do not stuck anymore but --enable-jails is still very very slow and takes about 10 minutes.

Please investigate further. I can provide all additional information.

File: fail2ban/server/filter.py (hunk 1 above)
Comment (yarikoptic) [inline fail2ban/server/filter.py]: so agreed that no OrderedDict here? ;)

**Comment by sebres** [comment]
all wishes are done, test cases passed, so merge now...

**Comment by yarikoptic** [comment]
Awesome! Thanks!

**Pull request merged**